zrml/prediction-markets/src/lib.rs

```diff
@@ -1491,17 +1511,39 @@ mod pallet {
             disputes: &[MarketDispute<T::AccountId, T::BlockNumber>],
             report: &Report<T::AccountId, T::BlockNumber>,
             outcome: &OutcomeReport,
+            dispute_mechanism: &MarketDisputeMechanism<T::AccountId>,
         ) -> DispatchResult {
             if let Some(last_dispute) = disputes.last() {
-                ensure!(&last_dispute.outcome != outcome, Error::<T>::CannotDisputeSameOutcome);
+                match dispute_mechanism {
+                    MarketDisputeMechanism::GlobalDisputes => {
+                        for d in disputes {
+                            ensure!(&d.outcome != outcome, Error::<T>::CannotDisputeSameOutcome);
+                        }
+                    }
+                    MarketDisputeMechanism::SimpleDisputes
+                    | MarketDisputeMechanism::Authorized(_)
+                    | MarketDisputeMechanism::Court => {
+                        ensure!(
+                            &last_dispute.outcome != outcome,
+                            Error::<T>::CannotDisputeSameOutcome
+                        );
+                    }
+                };
             } else {
```

Comment:
For `SimpleDisputes` it's clear here, that the last dispute should not match the one which is new. But Authorized has no effect as explained above / below. And for court I have no idea, if we should compare the outcome of the last dispute.